NoBeardProject/src/main/java/machine/NoBeardMachine.java

```diff
@@ -89,9 +94,37 @@ public void loadProgram(int atAddr, byte[] data) {
     public void runProgram(int startPc) {
         System.out.println("Starting programm at pc " + startPc);
         controlUnit.startMachine(startPc);
-        while (getState() == ControlUnit.MachineState.RUNNING) {
-            step();
+        breakpointIndex = 0;
+        jumpToNextBreakpoint();
+    }
+
```

Comment:
jumpttoNextBreakpoint() is maybe a bit misleading. Wouldn't it be better to call it runUntilNextBreakpoint()?